pkg/controller/recommendation/recommendation_trigger_controller.go

```
@@ -0,0 +1,160 @@
+	if newStatus.Recommendations[currentMissionIndex].Message != "Success" {
+		err = c.Client.Delete(context.TODO(), recommendation)
+		if err != nil {
+			klog.Errorf("Failed to delete recommendation %s when checking: %v", klog.KObj(recommendation), err)
+		}
+	}
+
+	updateRecommendationRuleStatus(context.TODO(), c.Client, c.Recorder, recommendationRule, newStatus)
+
+	return ctrl.Result{}, nil
+}
+
+func (c *RecommendationTriggerController) SetupWithManager(mgr ctrl.Manager) error {
+	c.discoveryClient = discovery.NewDiscoveryClientForConfigOrDie(mgr.GetConfig())
+	c.dynamicClient = dynamic.NewForConfigOrDie(mgr.GetConfig())
+
+	controller, err := controller.New("recommendation-trigger-controller", mgr, controller.Options{
+		Reconciler: c})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to Recommendation that runNumber decrease
+	return controller.Watch(&source.Kind{Type: &analysisv1alpha1.Recommendation{}}, &recommendationEventHandler{
+		enqueueHandler: handler.EnqueueRequestForObject{},
+	})
+}
+
+type recommendationEventHandler struct {
+	enqueueHandler handler.EnqueueRequestForObject
+}
+
+func (h *recommendationEventHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+}
+
+func (h *recommendationEventHandler) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+}
+
+func (h *recommendationEventHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	newRecommendation := evt.ObjectNew.(*analysisv1alpha1.Recommendation)
+	oldRecommendation := evt.ObjectOld.(*analysisv1alpha1.Recommendation)
+	klog.V(6).Infof("recommendation %s OnUpdate", klog.KObj(newRecommendation))
+
+	oldRunNumber, _ := utils.GetRunNumber(oldRecommendation)
+	newRunNumber, _ := utils.GetRunNumber(newRecommendation)
```

the error is not handled

---

if get run number failed , we treat it as a empty number which by default is 0.